easybuild/easyblocks/b/blender.py

```diff
@@ -0,0 +1,97 @@
+##
+# Copyright 2009-2018 Ghent University
+#
+# This file is part of EasyBuild,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://www.vscentrum.be),
+# Flemish Research Foundation (FWO) (http://www.fwo.be/en)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# https://github.com/easybuilders/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for Blender, implemented as an easyblock
+
+@author: Stijn De Weirdt (Ghent University)
+@author: Dries Verdegem (Ghent University)
+@author: Kenneth Hoste (Ghent University)
+@author: Pieter De Baets (Ghent University)
+@author: Jens Timmerman (Ghent University)
+@author: Ward Poelmans (Ghent University)
+@author: Samuel Moors (Vrije Universiteit Brussel)
+"""
+import glob
+import os
+
+from easybuild.easyblocks.generic.cmakemake import CMakeMake
+from easybuild.tools.modules import get_software_root, get_software_version
+from easybuild.tools.systemtools import get_shared_lib_ext
+
+
+class EB_Blender(CMakeMake):
+    """Support for building Blender."""
+
+    def configure_step(self):
+        """Set CMake options for Blender"""
+        self.cfg['separate_build_dir'] = True
+
+        self.cfg.update('configopts', '-DWITH_INSTALL_PORTABLE=OFF')
+        self.cfg.update('configopts', '-DWITH_BUILDINFO=OFF')
+
+        # disable SSE detection to give EasyBuild full control over optimization compiler flags being used
+        self.cfg.update('configopts', '-DWITH_CPU_SSE=OFF')
+        self.cfg.update('configopts', '-DCMAKE_C_FLAGS_RELEASE="-DNDEBUG"')
+        self.cfg.update('configopts', '-DCMAKE_CXX_FLAGS_RELEASE="-DNDEBUG"')
+
+        # these are needed until extra dependencies are added for them to work
+        self.cfg.update('configopts', '-DWITH_GAMEENGINE=OFF')
+        self.cfg.update('configopts', '-DWITH_SYSTEM_GLEW=OFF')
```

Comment:
@smoors Ideally, we avoid hardcoding these in the easyblock. It's OK to do this in an easyconfig file, since if you don't want these you can easily get around it by tweaking the easyconfig.

You could do this with constructions like this, for example:

```python
if '-DWITH_SYSTEM_GLEW=' not in self.cfg['configopts']:
    self.cfg.update('configopts', '-DWITH_SYSTEM_GLEW=OFF')
```

Even better, use a loop for all of these, over a map, something like this (untested):

```python
default_config_opts = {
    'WITH_INSTALL_PORTABLE': 'OFF',
    ...
}
for key in sorted(default_config_opts):
    opt = '-D%s=' % key
    if opt not in self.cfg['configopts']:
        self.cfg.update('configopts', opt + default_config_opts[key])